src/System.IO.Pipelines/tests/StreamPipeWriterTests.cs

```diff
@@ -122,6 +126,7 @@ static async Task DoReadsAsync(PipeReader reader, byte[][] reads)
                     Assert.Equal(reads[index], buffer.ToArray());
                     reader.AdvanceTo(buffer.End);
                     index++;
+                    waitForRead?.TrySetResult(null);
```

Comment:
Why do you need a nullable check here? Shouldn't it always be set by the time you hit this?

Reply:
Yes, I'll remove it.